elastic/kibana
# [SecuritySolution][AssetCriticality] Add check that prevents UI from calling API when feature is disabled

## Summary

Prevent the asset criticality UI from calling privileges API when the feature is disabled.


* The Asset Criticality UI should check if the advanced setting is enabled and only then make API requests to fetch the privileges.

* The Asset Criticality UI should check if the user has `appfeatures.advancedInsights` enabled before making API calls to the priviles APIs. 


### How to test it?
* Enable/disable the asset criticality advanced setting 
* Check if the UI is calling the privileges API

* On serverless, log in with a non-admin user on the essentials tier
  * On dev mode, you could also delete the advancedInsigh app feature [here](https://github.com/machadoum/kibana/blob/57374ab80d5aa7d53ffa1e88d52e9c184d796e02/x-pack/plugins/security_solution_serverless/common/pli/pli_config.ts#L20)
* Check if the UI is calling the privileges API



### Checklist
